spec/lib/efile/nj/nj1040_spec.rb

```diff
@@ -0,0 +1,97 @@
+require 'rails_helper'
+
+describe Efile::Nj::Nj1040 do
+  let(:intake) { create(:state_file_nj_intake) }
+  let(:instance) do
+    described_class.new(
+      year: MultiTenantService.statefile.current_tax_year,
+      intake: intake
+    )
+  end
+
+  context 'when filing status is single' do
+    it "sets line 6 to 1000" do
+      instance.calculate
+      expect(instance.lines[:NJ1040_LINE_6].value).to eq(1000)
+    end
+
+    context 'when filer is older than 65' do
+      before do
+        intake.primary_birth_date = Date.new(1900, 1, 1)
+        instance.calculate
+      end
+      it 'checks the self 65+ checkbox and sets line 7 to 1000' do
+        expect(instance.lines[:NJ1040_LINE_7_SELF].value).to eq(true)
+        expect(instance.lines[:NJ1040_LINE_7_SPOUSE].value).to eq(false)
+        expect(instance.lines[:NJ1040_LINE_7].value).to eq(1000)
+      end
+    end
+
+    context 'when filer is younger than 65' do
+      before do
+        intake.primary_birth_date = Date.new(2000, 1, 1)
```

Comment:
[pebble] same comment as above that 1960 should be the year under test for younger than 65.

Reply:
Original concern was that if we chose 1957 for younger than 65, next year that would be OLDER than 65 and the test would break. I can fix this by making it dynamically generate the birthdate.

Reply:
Hm. I'm not sure. It feels weird to just recreate the logic of our code in our tests (`current_year - 65` or whatever) but I also see your point about it breaking next year. 

I think I prefer the dynamic to the set-it-to-2000 approach, but I wonder if there's any other option. Hm.